lib/Runtime/Library/RuntimeFunction.h

```diff
@@ -18,10 +18,13 @@ namespace Js
 
         void SetFunctionNameId(Var nameId);
 
-        // this is for cached source string for the function. Possible values are:
+        // This is for cached source string for the function. Possible values are:
         // NULL; initialized for anonymous methods.
         // propertyId in Int31 format; this is used for fastDOM function as well as library function
         // JavascriptString: composed using functionname from the propertyId, or fixed string for anonymous functions.
+        // NOTE: This has a side-effect that after toString() is called for the first time on a built-in function the functionNameId gets replaced with a string like "function foo() { native code }".
```

Comment:
I added it so that anyone using this field in future is aware of this behavior and can avoid a bug like we are currently fixing in debugger.

---
In reply to: [129430780](https://github.com/Microsoft/ChakraCore/pull/3429#discussion_r129430780) [](ancestors = 129430780)